app/lib/advanced_text_formatter.rb

```diff
@@ -98,7 +98,8 @@
   private
 
   def format_markdown(html)
-    html = markdown_formatter.render(html)
+    # Force escape usernames in mentions before formatting
+    html = markdown_formatter.render(html.gsub(Account::MENTION_RE) { |re| re.gsub('_', '\\_') })
```

Comment:
`MENTION_RE` doesn't match backslashes, so `re` cannot contain escaped underscores.
The reason escaping here works is because redcarpet converts escaped underscores to underscores instead of u elements.
I haven't looked into why manually escaping usernames doesn't produce a mention, but I think that's expected behaviour.